abstractions/ruby/microsoft_kiota_abstractions/lib/microsoft_kiota_abstractions/authentication/access_token_provider.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+require 'concurrent'
+require 'oauth2'
+require_relative 'extensions/oauth2_ext'
+require_relative 'allowed_hosts_validator'
+require_relative 'contexts/client_credential_context'
+require_relative 'contexts/authorization_code_context'
+require_relative 'contexts/on_behalf_of_context'
+
+module MicrosoftKiotaAbstractions
+  # Access Token Provider class implementation
+  class AccessTokenProvider
+    include Concurrent::Async
+    # This is the initializer for AccessTokenProvider.
+    # :params
+    #   token_request_context: a instance of one of our token request context
+    #   allowed_hosts: an array of strings, where each string is an allowed host, default is empty
+    #   scopes: an array of strings, where each string is a scope, default is empty array 
+    #   auth_code: a string containting the auth code; default is nil, can be updated post-initialization
+    def initialize(token_request_context, allowed_hosts = [], scopes = [])
```

Comment:
I believe the first parameter needs to be removed and the implementation will call this base constructor

I also believe it should instantiate the host validator instead of throwing a not implemented exception